File: Backend/AstroGame.Shared/Models/Buildings/DynamicBuildingCost.cs
using AstroGame.Shared.Models.Technologies;

namespace AstroGame.Shared.Models.Buildings
{
    public class DynamicBuildingCost : TechnologyCost
    {
        /// <summary>
        /// The base value for the calculation
        /// </summary>
        public double BaseValue { get; set; }

        /// <summary>
        /// The multiplier for the calculation
        /// </summary>
        public double Multiplier { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/buildings/dynamic_building_cost.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/resources/dynamic_cost.dart';
import 'package:astrogame_app/models/technologies/technology_cost.dart';
import 'package:json_annotation/json_annotation.dart';

part 'dynamic_building_cost.g.dart';

@GuidConverter()
@JsonSerializable()
class DynamicBuildingCost extends TechnologyCost with DynamicCost {
  @JsonKey()
  late double baseValue;

  @JsonKey()
  late double multiplier;

  DynamicBuildingCost();

  factory DynamicBuildingCost.fromJson(Map<String, dynamic> json) =>
      _$DynamicBuildingCostFromJson(json);
  Map<String, dynamic> toJson() => _$DynamicBuildingCostToJson(this);
}
